src/helpers/pull-helpers/pull-request-parsing.ts

```diff
@@ -0,0 +1,105 @@
+import { encode } from "gpt-tokenizer";
+import { TokenLimits } from "../../types/llm";
+import { EncodeOptions } from "gpt-tokenizer/esm/GptEncoding";
+import { Context } from "../../types";
+
+export async function processPullRequestDiff(diff: string, tokenLimits: TokenLimits, logger: Context["logger"]) {
+  const { runningTokenCount, tokensRemaining } = tokenLimits;
+
+  // parse the diff into per-file diffs for quicker processing
+  const excludedFiles = ["bun.lockb"];
```

Comment:
This actually should be dynamically populated from gitignore and [linguist-generated](https://docs.github.com/en/repositories/working-with-files/managing-files/customizing-how-changed-files-appear-on-github).